docs/DeveloperGuide.md

```diff
@@ -1318,6 +1318,91 @@ testers are expected to do more *exploratory* testing.
+    1. Test case: `add-i in/Chocolate iq/4`
+       <br>Expected: No ingredient is added. Error details shown in the result display box.
+    1. Test case: `add-i iq/4 iu/bars`
+       <br>Expected: No ingredient is added. Error details shown in the result display box.
+    1. Test case: `add-i in/Chocolate iu/bars`
+       <br>Expected: No ingredient is added. Error details shown in the result display box.
+    1. Test case: `add-i in/Chocolate iq/0 iu/bars`
+       <br>Expected: No ingredient is added. Error details shown in the result display box.
+    1. Test case: `add-i`
+       <br>Expected: No ingredient is added. Error details shown in the result display box.
+
+1. Deleting an ingredient
+    1. Prerequisites: Inventory has at least 1 ingredient and at most 3 ingredients.
+    1. Test case: `delete-i 1`
+       <br>Expected: First ingredient in inventory is deleted.
+       If initially on Home tab, switch to Inventory & Statistics tab. Details of deleted ingredient will be shown in the result display box.
+    1. Test case: `delete-i 0`
+       <br>Expected: No ingredient is deleted. Error details shown in the result display box.
+    1. Test case: `delete-i abc`
+       <br>Expected: No ingredient is deleted. Error details shown in the result display box.
+    1. Test case: `delete-i`
+       <br>Expected: No ingredient is deleted. Error details shown in the result display box.
+
+1. Editing an ingredient
+    1. Prerequisites: Inventory only has these 3 ingredients:
+        * name: Apple, unit: whole
+        * name: Banana, unit: whole
+        * name: Carrot, unit: pack
+    1. Test case: `edit-i 1 in/Rice iq/4 iu/bags`
+       <br>Expected: First ingredient in the inventory is edited to have ingredient name 'Rice', quantity '4' and unit 'bags'.
+       If initially on Home tab, switch to Inventory & Statistics tab.
+       Its position in the inventory may change. Details of the edited ingredient are shown in the result display box.
+    1. Test case: `edit-i 3 in/Apricot`
+       <br>Expected: Third ingredient in the inventory is edited to have ingredient name 'Apricot'.
+       If initially on Home tab, switch to Inventory & Statistics tab.
+       Its position in the inventory may change. Details of the edited ingredient are shown in the result display box.
+    1. Test case: `edit-i 0 in/Apple`
+       <br> Expected: No ingredient is edited. Error details shown in the result display box.
+    1. Test case: `edit-i in/Apple`
+       <br> Expected: No ingredient is edited. Error details shown in the result display box.
+    1. Test case: `edit-i 1`
+       <br> Expected: No ingredient is edited. Error details shown in the result display box.
+
+1. Finding ingredients by keywords
+    1. Prerequisites: There are exactly 2 ingredients in the inventory.
+        * name: Green Apple, quantity: 10, unit: whole
+        * name: Banana, quantity: 13, unit: whole
+        * name: Carrot, quantity: 20, unit: pack
```

Comment:
It should be Inventory only has these 3 ingredients in line 1372 like Editing ingredient section